### PR TITLE
Nerfs most modsuits by giving them all armour boosters

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request

All security-grade modsuits (security, safeguard, magnate, blueshield, ERT, etc) have been outfitted with armour booster modules, built in and non-removable.

"But wait, Yakumo, that's a huge buff, isn't it?"

haha no

In """"compensation"""" for this built-in module, I have halved the armour values on all applicable modsuits, and moved said values to the armour booster modules. Every modsuit has its own specific booster with the moved armour values built in.

Also almost all of them don't get the speed boost, because it's just a poor copy of the syndicate one. I made an exception for the HoS MOD however because the secoff modsuits were recently buffed to have almost the same armour (we love power creep)

image
 
Contains an early pull of https://github.com/tgstation/tgstation/pull/80680

## Why It's Good For The Game

EVA combat is incredibly goofy right now. Armour on MODsuits was originally stripped out severely because having armour in already awful space combat was a little bit shit, and modsuits aren't supposed to be a be-all-end-all of protective drip. Except the Syndicate who got shafted by having dogshit armour in space, because most of their armour is in the armour booster.

Now everyone is on the same playing field again. Want to have actual armour in a MODsuit? You're gonna have to sacrifice your EVA capability to do so. No more free rides, and the syndicate is a respective threat in comparison in the same EVA environment again.

## Changelog

:cl:
add: Security, HoS, Captain, Blueshield, and Centcom roles have had armour booster modules built into their modsuits. Most do not reduce the slowdown of your suit (except the HoS one)
balance: Armour values in said modsuits have been halved across the board, with the lost values moved to the armour boosters.
balance: Effectively, armour in EVA combat is severely nerfed for everyone except the syndicate who already had armour boosters.
/:cl:

### CODE REVIEW

**Comment by projectkepler-ru** [comment]
This is actually an interesting idea, what do you think? @StrangeWeirdKitten

**Comment by YakumoChen** [comment]
I actually asked for their input before I made the PR, since the alternative was just buffing the Syndicate MODsuit

**Comment by projectkepler-ru** [comment]
we'll wait a bit

**Comment by projectkepler-ru** [comment]
And how does this work for a modsuit that get created in robotics?

**Comment by YakumoChen** [comment]
its built into the MOD theme, not just a pre-equipped module. It is installed the moment you complete the suit.

File: modular_skyrat/modules/blueshield/code/modsuit/mod_type.dm
Comment (Majkl-J) [inline modular_skyrat/modules/blueshield/code/modsuit/mod_type.dm:17]: Can be made modular with
```
/obj/item/mod/control/pre_equipped/blueshield/Initialize()
	default_pins += /obj/item/mod/module/armor_booster/nanotrasen/blueshield
	. = ..()
```

Whilst you're at it, you can do the same to the rest of the bubber edits in this file which I appear to have missed modularizing the codebase

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/blueshield/code/modsuit/mod_type.dm
Comment (YakumoChen) [reply modular_skyrat/modules/blueshield/code/modsuit/mod_type.dm]: I've done some of it, some I just let the override do its thing, and tried to clean up my code a bit to just have more of these Initialize procs instead of overriding to clean the code a bit better, let me know if there's any other way to improve it

**Comment by Detective-Google** [comment]
something I'd note about this is that if you spend a lot of time observing a lot of players already don't understand the armor booster system and don't use them when playing contractor or buying a blood-red, I have some concern that this may lead to inexperienced players having a poor time due to insufficient signposting of the system and how it works.

**Comment by YakumoChen** [comment]
It does come with a new button on the hotbar, and it shows up in the module list with a (admittedly dense) complete description. The most players could miss out of not fully knowing the change is the reduction in armour, as the default mode is EVA-friendly.

**Comment by YakumoChen** [comment]
at worst this will just encourage the playerbase to learn about the armour booster as station roles so they can take advantage of it when playing antag as well

A test merge might also be good just to gauge things however and bring attention to the PR.

**Review by Majkl-J — APPROVED** [review]
The actual number changes are not something I'll comment on since balance questions go beyond me usually. But code looks solid, so does modularity

**Comment by nevimer** [comment]
Blocked by me multiple times in the past. Under no circumstances will I accept MORE modsuit action button bloat dependence. For players, it's difficult to tell what mode a modsuit is in at a glance, if it's space worthy, etc. It's AWFUL TG design and we should move things to being simpler and clear instead of trying to enforce some weird dota-esque system where you have to micromanage 50 action buttons.

**Comment by YakumoChen** [comment]
should I just buff the syndicate modsuit then?

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]
The biggest thing is action button and icon issues. The armor module is not communicative, and that's just a simple sprite thing. I think a good solution, that is also easy to make, would be to give it a pre-existing sprite like the SWAT armor which will communicate clearly what it does at a glance. You'll need to modularize the sprite override.

image

File: modular_zubbers/code/datums/armor_overrides.dm
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/datums/armor_overrides.dm:85]: No, I do not want EVA armor to be good at all, even for antags. The idea is to pull the fight out of space, not incentivize antags fighting only in space.

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]
Decrease the armor across all of the modsuits. You can pretty much comment out all of Skyrat's armor overrides for the modsuits while giving them modules that buff them to what they are now inside atmosphere. I don't want antags to have good armor in space while everyone else has shit armor. I want it so if you breach the station, you're going to be fighting in those breached areas with poor armor.

**Comment by StrangeWeirdKitten** [comment]
I'm saying this purely because I don't see code changes that touch any base modsuit armor. You're making it so the captain's modsuit will get 20 added armor with the boost. Which is something I doubt you want.

**Comment by YakumoChen** [comment]
> I'm saying this purely because I don't see code changes that touch any base modsuit armor. You're making it so the captain's modsuit will get 20 added armor with the boost, making it 60. Which is something I doubt you want.

It seems to override the values just fine per my screenshot in the OP, and the code changes are right there in the armour overrides file?

**Comment by YakumoChen** [comment]
oh i see, I overwrote the file with the wrong stuff

File: modular_zubbers/code/datums/armor_overrides.dm
Comment (YakumoChen) [reply modular_zubbers/code/datums/armor_overrides.dm]: this is from a bad commit while trying to make the other PR, ignore it

**Comment by YakumoChen** [comment]
@StrangeWeirdKitten I have changed the icons, reverted the accidental commit, and added balloon alerts so users will know exactly what it does when (de)activated

**Review by StrangeWeirdKitten — APPROVED** [review]

**Comment by StrangeWeirdKitten** [comment]
I'll find a point to test merge this

**Comment by Moostard** [comment]
I dislike the changes.

Being able to remove the slowdown and get more armor is a really good thing while also having a powerful kit (Good cell, better storage, EMP shield, ion/adv. ion thrusters).
Meanwhile security suits do not even have ion jetpacks.

Crew will be punished for forgetting their armor booster on station to not have paper armor and if the module takes space, also will limit the possible modifications. Not to mention this would not give any speed boost.

Space combat has reputation for being unfun, since people move at speed of light and dodging projectiles cannot be any easier.
"Discouraging" space combat by lower armor in space seems to be better alternative than promoting it.
If it is such a big problem, making syndicate mods not losing armor in space would be much better imo.

**Comment by YakumoChen** [comment]
> Being able to remove the slowdown and get more armor is a really good thing while also having a powerful kit (Good cell, better storage, EMP shield, ion/adv. ion thrusters).
Meanwhile security suits do not even have ion jetpacks.

The slowdown and lower armour in EVA are handicaps for antags when security have had their MODs vastly buffed (past old hardsuit levels)
Ion jetpack can be bought from cargo for 100cr at round start

> Crew will be punished for forgetting their armor booster on station to not have paper armor and if the module takes space, also will limit the possible modifications. Not to mention this would not give any speed boost.

So just like antags? You're not encouraged to wear MODsuits on station. Why would I give a speed boost when they station MODs have it before?
Keep in mind armour boosters are unobtainable outside of built into MODs

> Space combat has reputation for being unfun, since people move at speed of light and dodging projectiles cannot be any easier.
"Discouraging" space combat by lower armor in space seems to be better alternative than promoting it.
If it is such a big problem, making syndicate mods not losing armor in space would be much better imo.

Dragging out space combat by taking twice as much ammo to do anything is not going to happen

**Comment by YakumoChen** [comment]
Was briefly nonmodular and made modular by https://github.com/tgstation/tgstation/pull/80680

**Comment by StrangeWeirdKitten** [comment]
Test merging

**Comment by StrangeWeirdKitten** [comment]
I peered into the armor changes and we've been test merging it. This is 100 percent fine. There's plans for improvements.

**Comment by Ashrollen** [comment]
> The syndicate modsuit isn't designed to be easily upgraded and is antag gear. Security can easily modify theirs at science. Non-issue.

No primary access to science. No willing scientist. All burden falls on willing to play science team which we know usually goes god knows where. Making security rely on other departments to be fair had prooven itself as worst idea ever. Moment i hear each time "But station has ..." makes me tremble in shock fear and mostly hatred because never i saw any other department depend so much on others non-troll behaviour than this one. And trust me there will be trolls many trolls that will do anything but help. 



> The slowdown and lower armour in EVA are handicaps for antags when security have had their MODs vastly buffed (past old hardsuit levels)
> Ion jetpack can be bought from cargo for 100cr at round start

Again in terms of slowdown/armor and such termins you must first check what exact point of low complexity of such suit. It was made to be fast for trade of complexity. Now its slow has low armor and complexity near hard suit which makes any player from security to consider jetpack and standart EVA suit as better option.

Another case of negligence since all goes to cargo and overdependance of said department. I must go to cargo and hope for workers there to not fluoride stare at me or waste my time or just straight away hide case with order (Was prooven to be constant act and is considered IC issue by adminstration which means nobody but cargo itself can change it). Its as previously was mentioned issue of person that got used to "Ideal state of things". There is no ideal state of things on station period.

**Comment by YakumoChen** [comment]
> 
> No primary access to science. No willing scientist. All burden falls on willing to play science team which we know usually goes god knows where. Making security rely on other departments to be fair had prooven itself as worst idea ever. 
>  (the rest of these similar quotes)

Irrelevant opinions. Station roles are designed around cooperation for progression and generally assume competent and willing players to cooperate with. If you cannot trust other players to help make your job easier, or are not capable of interacting without making a massive ass of yourself this server and codebase is not for you.

> Again in terms of slowdown/armor and such termins you must first check what exact point of low complexity of such suit. It was made to be fast for trade of complexity. Now its slow has low armor and complexity near hard suit which makes any player from security to consider jetpack and standart EVA suit as better option.

You do not understand the nature of the change. 

Combat hardsuits now switch between "halved armour, EVA capable" and "full armor, no EVA". 

Only the HoS and antag modsuits have speed benefits in full armour mode

This does not affect MODsuit complexity in any way. 

This does not change antag modsuits in any way, only security, blueshield, and captain modsuit are changed.

**Comment by Ashrollen** [comment]
You tested it i suppose only on one type antagonist. I must congratulate
you your idea yesterday was tested by average space dragon. Entire command
and security annihilated due to choice of either armor or eva in a room.
But i suppose it was your intention that any person that now spaces room
and somewhat immune to pressure and temperature will have severe advantages.

сб, 20 янв. 2024 г., 06:28 YakumoChen ***@***.***>:

**Comment by YakumoChen** [comment]
> You tested it i suppose only on one type antagonist. I must congratulate you your idea yesterday was tested by average space dragon. Entire command and security annihilated due to choice of either armor or eva in a room. But i suppose it was your intention that any person that now spaces room and somewhat immune to pressure and temperature will have severe advantages

Somehow on TG they can deal with these kinds of PMS antags just fine, with even worse armour on MODsuits than we still get *after* this nerf.

I've seen space dragons erase the entire crew even before this PR was tested or merged.

This is simply a case of the crew having a lack of skill.

**Comment by Majkl-J** [comment]
> > You tested it i suppose only on one type antagonist. I must congratulate you your idea yesterday was tested by average space dragon. Entire command and security annihilated due to choice of either armor or eva in a room. But i suppose it was your intention that any person that now spaces room and somewhat immune to pressure and temperature will have severe advantages
> 
> Somehow on TG they can deal with these kinds of PMS antags just fine, with even worse armour on MODsuits than we still get _after_ this nerf.
> 
> I've seen space dragons erase the entire crew even before this PR was tested or merged.
> 
> This is simply a case of the crew having a lack of skill.
> 
> This was test merged for two weeks and you have been the only person to voice a complaint about the balance.
> 
> Get good.

You can't expect the crew to be universally skilled in combat, beside we have a overall lower skill ceiling. The issue with dragon winning before was it spawning during lowpop when it shouldn't have been doing so, which I believe has been fixed.

Now it smacked a fully equipped crew mostly because of the fact that it spaced the station as it went, making combat with it basically impossible for almost anyone, aside from like 2 miners who couldn't do much.

Anyhow there have been several complaints on the discord so if this trend continues adjustments will have to be made.

**Comment by YakumoChen** [comment]
If that's the case, that's fine, I'm just irritated with this one guy making repeated comments displaying a clear lack of knowledge on the subject, I should probably just block them tbh

**Pull request merged**
